woocommerce/woocommerce-android
# Issue/4579 onboarding/welcome dialog

Closes: #4579


### Description


This PR introduces a welcome dialog which is shown w

hen the user taps on "Collect Payment" button for the first time (once per site) - the user gets navigated to the onboarding flow. It is not shown if the user completes the onboarding in the app settings before tapping on the collect payment button.

| Light | Dark |
|--|--|
| <img src="https://user-images.githubusercontent.com/2261188/135290631-c7060873-3e54-4510-9c60-268b7575c65b.png" width="250px" /> | <img src="https://user-images.githubusercontent.com/2261188/135290337-668e8874-b9e0-472c-83e7-83b766feeb9e.png" width="250px" /> |


@woocommerce/mobile-designers I used "Continue" button instead of "Open settings" button in Figma. Since we technically don't redirect the user to the settings I found it a bit confusing, wdyt?


### Testing instructions

1. Open an order eligible for in person payments
2. Tap on "Collect Payment" button
3. Notice the welcome dialog is shown
4. Tap on continue
5. Notice the onboarding flow is shown -> if the site is onboarded -> the flow returns back to the order detail and starts scanning for readers
6. Cancel the flow
7. Tap on Collect Payment button again
8. Notice the welcome dialog is not shown again and the scanning starts right away
9. Switch to a different site, repeat steps 1-6 
10. Switch back to the original site and verify that the dialog is not shown

### Images/gif
